spec/system/planning_applications/replacement_document_validation_request_spec.rb

```diff
@@ -466,6 +470,68 @@
         expect(page).to have_content("Planning application has already been validated")
       end
     end
+
+    it "allows new replacement requests" do
+      click_link "Check and assess"
+      click_button "Documents"
+      click_link "Request replacement"
+
+      fill_in "List all issues with the document.", with: "This is very invalid"
+      click_button "Send request"
+
+      expect(page).to have_content("Replacement document validation request successfully created.")
+
+      document1.reload
+      expect(document1.replacement_document_validation_request).to eq(ReplacementDocumentValidationRequest.last)
+      expect(document1.replacement_document_validation_request.post_validation).to be true
+    end
+
+    it "appears in the post validation requests table" do
+      click_link "Check and assess"
+      click_button "Documents"
+      click_link "Request replacement"
+
+      fill_in "List all issues with the document.", with: "This is very invalid"
+      click_button "Send request"
+      click_link "Application"
+      click_link "Review non-validation requests"
+
+      within ".validation-requests-table" do
+        expect(page).to have_content(document1.name)
+      end
+    end
+
+    it "allows new replacement requests to be responded to" do
+      click_link "Check and assess"
+      click_button "Documents"
+      click_link "Request replacement"
+
+      fill_in "List all issues with the document.", with: "This is very invalid"
+      click_button "Send request"
+
+      expect(page).to have_content("Replacement document validation request successfully created.")
+      document1.reload
+
+      document2 = create(:document, :with_file, planning_application: planning_application)
+      request = planning_application.replacement_document_validation_requests.last
+      request.new_document = document2
+      request.state = "closed"
+      request.save!
+      request.old_document.archive("replaced by new document")
```

Comment:
I'm a little wary of this section: I'm not sure if there's a better way to generate a response, but I'm concerned that manipulating the objects directly might just bypass any issues that might occur in real usage. However, this seems to be what other tests do already, and I've verified as best I can that it's not just working by accident. Since we can't AFAIK access bops-applicants directly during these tests I think this is the best we can do.